applications/FluidDynamicsApplication/custom_constitutive/fluid_adjoint_constitutive_law.h

```diff
@@ -0,0 +1,132 @@
+//    |  /           |
```

Comment:
Just wondering if it makes sense to directly add these methods to the base constitutive law (the problem would be how to store the `mrPrimalConstitutiveLaw`...)